src/java.base/share/classes/java/lang/foreign/ValueLayout.java

```diff
@@ -180,126 +123,45 @@ public boolean equals(Object other) {
-    public ValueLayout withBitAlignment(long bitAlignment) {
-        return (ValueLayout)super.withBitAlignment(bitAlignment);
-    }
-
-    static void checkCarrierSize(Class<?> carrier, long size) {
-        if (!isValidCarrier(carrier)) {
-            throw new IllegalArgumentException("Invalid carrier: " + carrier.getName());
-        }
-        if (carrier == MemorySegment.class && size != ADDRESS_SIZE_BITS) {
-            throw new IllegalArgumentException("Address size mismatch: " + ADDRESS_SIZE_BITS + " != " + size);
-        }
-        if (carrier.isPrimitive()) {
-            int expectedSize =  carrier == boolean.class ? 8 : Wrapper.forPrimitiveType(carrier).bitWidth();
-            if (size != expectedSize) {
-                throw new IllegalArgumentException("Carrier size mismatch: " + carrier.getName() + " != " + size);
-            }
-        }
-    }
-
-    static boolean isValidCarrier(Class<?> carrier) {
-        return (carrier.isPrimitive() && carrier != void.class)
-                || carrier == MemorySegment.class;
-    }
-
-    @Stable
-    private VarHandle handle;
-
-    @ForceInline
-    VarHandle accessHandle() {
-        if (handle == null) {
-            // this store to stable field is safe, because return value of 'makeMemoryAccessVarHandle' has stable identity
-            handle = Utils.makeSegmentViewVarHandle(this);
-        }
-        return handle;
-    }
+    ValueLayout withBitAlignment(long bitAlignment);
 
     /**
      * A value layout whose carrier is {@code boolean.class}.
      *
      * @since 19
      */
-    @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
-    public static final class OfBoolean extends ValueLayout {
-        OfBoolean(ByteOrder order) {
-            super(boolean.class, order, 8);
-        }
-
-        OfBoolean(ByteOrder order, long bitAlignment, Optional<String> name) {
-            super(boolean.class, order, 8, bitAlignment, name);
-        }
+    @PreviewFeature(feature = PreviewFeature.Feature.FOREIGN)
+    sealed interface OfBoolean extends ValueLayout permits ValueLayouts.OfBooleanImpl {
 
         @Override
-        OfBoolean dup(long bitAlignment, Optional<String> name) {
-            return new OfBoolean(order(), bitAlignment, name);
-        }
+        OfBoolean withName(String name);
 
         @Override
-        public OfBoolean withName(String name) {
-            return (OfBoolean)super.withName(name);
-        }
+        OfBoolean withBitAlignment(long alignmentBits);
 
         @Override
-        public OfBoolean withBitAlignment(long bitAlignment) {
-            return (OfBoolean)super.withBitAlignment(bitAlignment);
-        }
+        OfBoolean withOrder(ByteOrder order);
 
-        @Override
-        public OfBoolean withOrder(ByteOrder order) {
+        /**
+         * {@return an OfBoolean with the provided {@code order}} ;
+         *
+         * @param order the byte order to use
+         */
+        static OfBoolean of(ByteOrder order) {
```

Comment:
I think we should drop these static methods, they don't seem that useful given one can build from the constant. Further, the naming is a little awkward with two "of" e.g. `ValueLayout.OfBoolean.of(...)`.

Let's use the concrete implementations when creating `JAVA_BOOLEAN` etc.